hipchat.go

```diff
@@ -86,15 +86,22 @@ type ErrorResponse struct {
 type Client struct {
 	AuthToken string
 	BaseURL   string
+	ProxyURL  string
```

Comment:
I have implemented that http.newRequest and client.do for all GET, POST now only one thing i need to ask is 
where do we ask for proxy settings from user ? may i make method e.g.
 ```go 
c.Transport = hipchat.SetProxy{"<AUTH TOKEN>","<PROXY_URL:PROXY_PORT>"} 
or handle with 
c := hipchat.NewProxyClient("<AUTH TOKEN>","<PROXY_URL:PROXY_PORT>")
```
sorry for too many questions i am new to GoLang.